packages/enzyme-adapter-react-16.1/src/ReactSixteenOneAdapter.js

```diff
@@ -295,8 +295,12 @@ class ReactSixteenOneAdapter extends EnzymeAdapter {
         return instance ? toTree(instance._reactInternalFiber).rendered : null;
       },
       simulateError(nodeHierarchy, rootNode, error) {
+        const isErrorBoundary = el => (
+          el.componentDidCatch || (el.constructor && el.constructor.getDerivedStateFromError)
+        );
+
         const { instance: catchingInstance } = nodeHierarchy
-          .find(x => x.instance && x.instance.componentDidCatch) || {};
+          .find(x => x.instance && isErrorBoundary(x.instance)) || {};
```

Comment:
Also, rather than relying on `constructor`, let's pass in `x.instance, x.type` (and update the hierarchy code to pass it)